### PR TITLE
train loop for CNN classificator

### PR DESCRIPTION
**Pull request by AstrakhantsevaAA**

### CODE REVIEW

File: src/config.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import os
+import torch
+from pathlib import Path
+from dataclasses import dataclass
+
+
+@dataclass
+class SystemConfig:
+    root_dir: Path = Path(__file__).parent.parent
```
Comment (b0nce) [inline src/config.py]: [nit] `.parents[1]`

File: src/data_utils/dataset.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    ):
+        self.data_dir = data_dir
+        self.images = data_dir.rglob("*.npy")
+        self.images_dict = defaultdict()
+        for image in self.images:
+            self.images_dict[image.stem] = image
+        df = pd.read_csv(csv_path)
+        df = df[df["split"] == phase]
+        self.data = df if test_size <= 0 else df.iloc[:test_size]
+        self.data["filepath"] = self.data.loc[:, "uid"].apply(lambda x: self.images_dict[x])
```
Comment (b0nce) [inline src/data_utils/dataset.py]: `.map` or `.replace` is more correct way of doing this

File: src/data_utils/dataset.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        self.labels = self.data.loc[:, "severity"]
+
+        self.transform = Compose(
+            [
+                Resize(224, 224),
+                ToTensorV2(),
+            ]
+        )
+        self.augmentation = None
```
Comment (b0nce) [inline src/data_utils/dataset.py:81]: [nit] move to separate function
you can also add parametrization for different augmentation types, e.g. "weak": <few augs>, "medium": \<more augs>, "hard": \<even more augs>

Comment (b0nce) [reply src/data_utils/dataset.py:81]: I still haven't saw the data but pretty sure `fliplr`, `flipud` and `colorjitter` will be good additional choices for augmentations, consider adding them

File: src/nets/define_net.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+def define_net(
+    model_name: str = "resnest-18",
+    freeze_grads: bool = False,
+    outputs: int = net_config.outputs,
+    pretrained: bool = False,
+    weights: Optional[str] = None,
+):
+
+    if model_name == "resnet18":
```
Comment (b0nce) [inline src/nets/define_net.py:18]: Default is `resnet-18` and not `resnet18`, seems like this is not intended

Comment (AstrakhantsevaAA) [reply src/nets/define_net.py:18]: didn't notice! thanks

File: src/data_utils/dataset.py (hunk 3 above)
Comment (AstrakhantsevaAA) [reply src/data_utils/dataset.py:81]: yes, I just copy pasted this part from my previous pet project XD

**Pull request merged**